src/dotnet/commands/dotnet-migrate/ProjectJsonParser.cs

```diff
@@ -0,0 +1,81 @@
+﻿using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Cli
+{
+    /// <summary>
+    /// Parses select data from a project.json without relying on ProjectModel.
+    /// Used to parse simple information.
+    /// </summary>
+    internal class ProjectJsonParser
+    {
+        public static string SdkPackageName => "Microsoft.DotNet.Core.Sdk";
+
+        public string SdkPackageVersion { get; }
+
+        public ProjectJsonParser(JObject projectJson)
+        {
+            SdkPackageVersion = GetSdkPackageVersion(projectJson);
+        }
+
+        private string GetSdkPackageVersion(JObject projectJson)
+        {
+            var sdkPackageNode = SelectJsonNodes(projectJson, property => property.Name == SdkPackageName).First();
```

Comment:
I'm not sure this is going to work because of all the type weirdness and casting in JSON.Net but I'll give it a try.